cxx/src/lib/io/fileio.cc

```diff
@@ -83,6 +84,86 @@ long int fwrite_to_file(Seismogram& d, const string dir,const string dfile)
 
 	return(foff);
 }
+long int fwrite_to_file(Ensemble<TimeSeries>& de, const string dir,const string dfile)
+{
+	try{
+		FILE *fp;
+		long int foff;
+		string fname;
+		if(dir.length()>0)
+		  /* for expected context for use in python we will assume dir does not
+			have a trailing path separator so we always insert / */
+			fname = dir + "/" + dfile;
+		else
+		  /* Null as always in unix means use current directory*/
+			fname=dfile;
+		if((fp=fopen(fname.c_str(),"a")) == NULL)
+		  /* use the name of the overloaded parent instead of the actual function - intentional*/
+			throw MsPASSError("fwrite_to_file:  Open failed on file "+fname,ErrorSeverity::Invalid);
+	  /* Both fseek and ftell can fail in weird circumstances, but I intentionally
+		do not trap that condition as if either have issues I am quite sure
+		the fwrite will fail */
+		fseek(fp,0L,2);
+		foff = ftell(fp);
+		for (int i = 0; i < de.member.size(); ++i) {
+			TimeSeries& t = de.member[i];
+			if (fwrite((void *)t.s.data(), sizeof(double), t.npts(), fp) != t.npts())
+			{
+				fclose(fp);
+				throw MsPASSError("fwrite_to_file:  fwrite error to file " + fname, ErrorSeverity::Invalid);
+			}
+			/* We always set these 3 attributes in Metadata so they can be properly
+			saved to the database after a successful write.  Repetitious with Seismogram
+			but a function to do this would be more confusing that helpful */
+			t.put_string(SEISMICMD_dir, dir);
+			t.put_string(SEISMICMD_dfile, dfile);
+			t.put_long(SEISMICMD_foff, foff);
+		}	
+		fclose(fp);
+
+		return foff;
+	}catch(...){throw;};
+}
+long int fwrite_to_file(Ensemble<Seismogram>& de, const string dir,const string dfile)
+{
+	try{
+		FILE *fp;
+		long int foff;
+		string fname;
+		if(dir.length()>0)
+		  /* for expected context for use in python we will assume dir does not
+			have a trailing path separator so we always insert / */
+			fname = dir + "/" + dfile;
+		else
+		  /* Null as always in unix means use current directory*/
+			fname=dfile;
+		if((fp=fopen(fname.c_str(),"a")) == NULL)
+		  /* use the name of the overloaded parent instead of the actual function - intentional*/
+			throw MsPASSError("fwrite_to_file:  Open failed on file "+fname,ErrorSeverity::Invalid);
+	  /* Both fseek and ftell can fail in weird circumstances, but I intentionally
+		do not trap that condition as if either have issues I am quite sure
+		the fwrite will fail */
+		fseek(fp,0L,2);
+		foff = ftell(fp);
```

Comment:
This call to fseek is misplaced.   It belongs inside the loop over members.  This code as is will not work.  Each member's metadata will have the same foff == 0.